examples/siro_sandbox/app_states/app_state_fetch.py

```diff
@@ -8,126 +8,403 @@
+        if self._held_target_obj_idx is None:
+            self._try_grasp_remote()
+        else:
+            self._update_held_and_try_throw_remote()
+
+        (
+            walk_dir,
+            distance_multiplier,
+        ) = self._nav_helper.get_humanoid_walk_hints_from_remote_gui_input(
+            visualize_path=False
+        )
+
+        # If the humanoid is near enough to the VR user that it doesn't need to walk
+        # (distance_multiplier == 0.0), then try to reach to held object.
+        reach_pos = None
+        if (
+            self._held_target_obj_idx is not None
+            and distance_multiplier == 0.0
+        ):
+            # vr_root_pos, _ = self._sandbox_service.remote_gui_input.get_head_pose()
+            # humanoid_pos = self._get_gui_agent_translation()
+            # dist_threshold = 0.25
+            # if (vr_root_pos - humanoid_pos).length() < dist_threshold:
+            # reach_pos = self._get_target_object_position(self._held_target_obj_idx)
+            # distance_multiplier = 0.0  # disable walking, but allow rotation via walk_dir
+            reach_pos = self._get_target_object_position(
+                self._held_target_obj_idx
+            )
+
+        grasp_object_id = None
+        drop_pos = None
+        throw_vel = None
+
+        assert isinstance(self._gui_agent_ctrl, GuiHumanoidController)
+        self._gui_agent_ctrl.set_act_hints(
+            walk_dir,
+            distance_multiplier,
+            grasp_object_id,
+            drop_pos,
+            self._camera_helper.lookat_offset_yaw,
+            throw_vel=throw_vel,
+            reach_pos=reach_pos,
+        )
+
     def _update_grasping_and_set_act_hints(self):
+        if self._is_remote_active():
+            self._update_grasping_and_set_act_hints_remote()
+        else:
+            self._update_grasping_and_set_act_hints_local()
+
+    def _update_grasping_and_set_act_hints_local(self):
         drop_pos = None
         grasp_object_id = None
+        throw_vel = None
+        reach_pos = None
 
         if self._held_target_obj_idx is not None:
-            if self._sandbox_service.gui_input.get_key_down(
+            if self._sandbox_service.gui_input.get_key(GuiInput.KeyNS.SPACE):
+                if self._recent_reach_pos:
+                    # Continue reaching towards the recent reach position (the
+                    # original position of the grasped object before it was snapped
+                    # into our hand).
+                    reach_pos = self._recent_reach_pos
+                else:
+                    # Reach towards our held object. This doesn't make a lot of sense,
+                    # but it creates a kind of hacky reaching animation over time,
+                    # because as we reach, the object in our hand moves, affecting our
+                    # reach pose on the next frame.
+                    reach_pos = self._get_target_object_position(
```

Comment:
@xavierpuigf I was trying to preserve your existing reaching-on-throw behavior while refactoring a lot of code in here. I would suggest rethinking how we do reaching during a throw; this logic doesn't make sense as I explained in my comment.